### ISSUE TITLE
`usize` incorrectly inferred as implementing `Iterator` when calling `.max(…)` on it

### ISSUE THREAD

**Issue by dfoxfranke** [rust-lang/rust-analyzer#20086]
The following program goes cleanly through `cargo check`, but gets a spurious error from rust-analyzer:

```rust
pub trait Iterable<'a, T>
where
    T: ?Sized + 'a,
{
    type Iter: Iterator<Item = &'a T>;
}

pub struct UnionIter<'a, A, B, T>
where
    A: Iterable<'a, T>,
    B: Iterable<'a, T>,
    T: ?Sized,
{
    a: A::Iter,
    b: B::Iter,
}

impl<'a, A, B, T> Iterator for UnionIter<'a, A, B, T>
where
    A: Iterable<'a, T>,
    B: Iterable<'a, T>,
    T: 'a + ?Sized,
{
    type Item = &'a T;

    fn size_hint(&self) -> (usize, Option<usize>) {
        let (a_min, a_max) = self.a.size_hint();
        let (b_min, b_max) = self.b.size_hint();

        let min = a_min.max(b_min);
        let max = if let (Some(a_max), Some(b_max)) = (a_max, b_max) {
            a_max.checked_add(b_max)
        } else {
            None
        };

        (min, max)
    }

    fn next(&mut self) -> Option<Self::Item> {
        todo!()
    }
}
```

rust-analyzer shows an error at the line `let min = a_min.max(b_min);`. It incorrectly infers that `a_min` (a `usize`) implements `Iterator`, thinks that when I call `.max` I mean `Iterator::max` rather than `Ord::max`, and gives the error, `expected 0 arguments, found 1`.

**rust-analyzer version**: 0.3.2509-standalone
**rustc version**: 1.87.0 (17067e9ac 2025-05-09)
**editor or extension**: VSCode, extension 0.3.2509

**Comment by dfoxfranke** [comment]
Possibly duplicate or related to #17233 and/or #19752.

**Comment by flodiebold** [comment]
Duplicate of #10653.
